package/kernel/lantiq/ltq-deu/src/ifxmips_aes.c

```diff
@@ -251,23 +251,25 @@ void ifx_deu_aes (void *ctx_arg, u8 *out_arg, const u8 *in_arg,
 
     /* To handle all non-aligned bytes (not aligned to 16B size) */
     if (byte_cnt) {
-        aes->ID3R = INPUT_ENDIAN_SWAP(*((u32 *) in_arg + (i * 4) + 0));
-        aes->ID2R = INPUT_ENDIAN_SWAP(*((u32 *) in_arg + (i * 4) + 1));
-        aes->ID1R = INPUT_ENDIAN_SWAP(*((u32 *) in_arg + (i * 4) + 2));
-        aes->ID0R = INPUT_ENDIAN_SWAP(*((u32 *) in_arg + (i * 4) + 3));    /* start crypto */
+        u8 *input[16];
```

Comment:
I think you should zero initialize `input`. The `memcpy` will only fill it partly as far as I understood and then you will have some random data from the stack which you encrypt and leak. 

I am also unsure if this works at all, I think you have to do AES on 16 bytes blocks and it is not possible to do it on a half block. As you also only take half of the `output` I think it is not possible to decrypt the input again.

Reply:
Actually the uninitialized pointers were fixed by 035ecee, do you think I should merge it with the original commit?
As for the second comment, just encrypting 5 bytes and the rest junk and take the 5 encrypted bytes works also for the decrypt. It does not work for DES. XTS (Cipher text stealing) would not work if AES works like DES.
Like in the XEX picture in the [wikipedia article](https://en.wikipedia.org/wiki/Disk_encryption_theory) AES is byte position preserving and you could replace the red stuff just with junk and the last block would still be ok for encryption and decryption.

Reply:
Yes you are right when only the first bytes are valid and the rest is garbage it will also decrypt to the same.

Even in 035eceef2c3c9c70c1fadd6244d37b30b2b445f3 the `temparea` is not fully initialized before it is used.
The last bytes could contain whatever data was on the stack before. This would leak uninitialized data from the stack. 
You should initialize it before or do something like this:
```
memset(temparea + byte_cnt, 0x0, 16 - byte_cnt);
```

I would prefer if you also merge both commits.